apps/extension/src/routes/page/onboarding/generate.tsx

```diff
@@ -14,20 +14,74 @@ import { generateSelector } from '../../../state/seed-phrase/generate';
 import { usePageNav } from '../../../utils/navigate';
 import { PagePath } from '../paths';
 import { WordLengthToogles } from '../../../shared/containers/word-length-toogles';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { fetchBlockHeight } from '../../../hooks/full-sync-height';
+import { localExtStorage } from '../../../storage/local';
+import { sample } from 'lodash';
+
+const fetchBlockHeightWithFallback = async (endpoints: string[]): Promise<number> => {
+  if (endpoints.length === 0) {
+    throw new Error('All RPC endpoints failed to fetch the block height.');
+  }
```

Comment:
At the moment, think this error will bubble up. We should move this to zustand, capture error states, and display them in the UI.

Reply:
I moved the logic to the [zustand store](https://github.com/prax-wallet/web/pull/164/commits/4ecea54ab1deb4893e43da25b3cf37a18b902888#diff-47254f5f6d488f18262f49e33c899348386b370c99f0f9a7f1219160bf492196R20-R22) and propagated the error to the caller ([useEffect](https://github.com/prax-wallet/web/pull/164/commits/6ae261e2ea68362d90e5181fbe0c5b9f322ddf94#diff-0c7c024c9b3806be547cbd5b3d147f6d4c33e6a1e68814dda964ac666c8d606aR34) hook) which sets the error in the component's state. 

<img width="480" alt="Screenshot 2024-09-01 at 10 36 39 AM" src="https://github.com/user-attachments/assets/406c0c03-2b09-4007-9c6a-55c13b317be3">

Reply:
confirmed proper error propagation with react query

<img width="462" alt="Screenshot 2024-09-03 at 10 50 15 PM" src="https://github.com/user-attachments/assets/5675eb61-ef4f-463e-8b4b-fccb4c710637">